example_configs/sssd.md

```diff
@@ -0,0 +1,149 @@
+
+The following configuration was tested on sssd version 2.9.1. It may not work on older versions.
+
+To work properly SSSD also needs corret openldap-client configuration.
+
+## Packages
+
+Here are the packages needed in Oracle Linux 9:
+
+`openldap-clients openldap-devel sssd sssd-ldap`
+
+
+## Configure OpenLDAP client
+
+### Certificates - skip if you don't use LDAPS
+
+1. Place your certs in `/etc/openldap/certs`
+2. Run `openssl rehash /etc/openldap/certs`
+
+### Config
+
+Create or modify `/etc/openldap/ldap.conf`:
+```
+URI ldaps://LDAP_SERVER_URL # Use ldap:// if you don't use LDAPS
+BASE dc=example,dc=com
+TLS_CACERTDIR /etc/openldap/certs # Skip if you don't use LDAPS
+ssl on # Skip if you don't use LDAPS
+```
+
+### Verify
+
+To check if your config is correct you can use `ldapsearch`:
+
+```bash
+ldapsearch -x -D "uid=admin,ou=people,dc=example,dc=com" -w "${ADMIN_PASSWORD}" -b "ou=people,dc=example,dc=com"
+```
+
+The above command should return list of your users.
+
+## Prepare LLDAP
+
+SSSD requires some specific setup.
+
+All of your users need `uidnumber` and `gidnumber` - these are not generated by default in LLDAP.
+
+To create these user attributes you can use either the web GUI or [lldap-cli](https://github.com/Zepmann/lldap-cli), but webp GUI does not support setting them.
+
+To create these custom user attributes with [lldap-cli](https://github.com/Zepmann/lldap-cli):
+
+```bash
+# Login to your LLDAP server
+eval $(lldap-cli -D admin -w abcd1234 login)
+
+# Add uidnumber and gidnumber attributes
+lldap-cli schema attribute user add uidnumber integer
+lldap-cli schema attribute user add gidnumber integer
+```
+
+You must also manually set the above attributes for each user, eg. for user `admin` with `uidnumber` 2000 and `gidnumber` 2000:
+```bash
+lldap-cli user update set admin uidnumber 2000
+lldap-cli user update set admin gidnumber 2000
+```
+
+> [!WARNING]
+> To set each user's home directory, use `homeDirectory` attribute. It can be also simply hardcoded in SSSD (as shown below).
+
+To overwrite user's shell you can use `loginShell` attribute.
+  
+To make groups discoverable by SSSD you need to add `posixGroup` objectClass:
+
+```bash
+lldap-cli schema objectclass group add posixGroup
+```
+
+## Configure SSSD
+
+1. Configure System Services for SSSD:
+
+```bash
+authselect select sssd
+```
+
+2. Create or modify `/etc/sssd/sssd.conf`:
+
+```bash
+[domain/default]
+ldap_uri               = ldaps://LDAP_SERVER_URL # Use ldap:// if you don't use LDAPS
+ldap_default_bind_dn   = uid=admin,ou=people,dc=example,dc=com
+ldap_default_authtok   = ADMIN_PASSWORD
+ldap_search_base       = dc=example,dc=com
+ldap_tls_reqcert       = allow # Skip if you don't use LDAPS
+ldap_tls_cacertdir     = /etc/openldap/certs # Skip if you don't use LDAPS
+ldap_schema = rfc2307bis
+
+enumerate              = true
+cache_credentials      = True
```

Comment:
It's confusing that there are 2 different casing for "true"

Reply:
Well, they are both vaild :D. I will keep them lowercase though.